src/main/java/net/cosmocat/marketplace/database/dal/repository/CategoryRepository.java

```diff
@@ -0,0 +1,29 @@
+package net.cosmocat.marketplace.database.dal.repository;
+
+import net.cosmocat.marketplace.database.entity.Category;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public interface CategoryRepository extends JpaRepository<Category, Long> {
+
+    Optional<Category> findByName(String name);
+
+    boolean existsByName(String name);
+
+    List<Category> findByNameContainingIgnoreCase(String name);
+
+    @Query("SELECT DISTINCT c FROM categories c JOIN c.tags t WHERE t = :tag")
```

Comment:
I cannot give you +2 because that's basic queries, that you can do with general jpa.
but for attempt I will give you +1